pythonAPIClient/client.py

```diff
@@ -45,25 +45,149 @@ def __init__(self, clientId, clientSecret):
-        pass
+        url = Client.base_URL + 'apps/register/?owner=' + owner + \
+              '&permissionLevel=' + permissionLevel
+        header = self.header_builder()
+        response = requests.post(url, headers=header)
+        if response.status_code == 404:
+            raise ResourceNotFoundError(response.json()['message'])
+        elif response.status_code == 422:
+            raise FormatError(response.json()['message'])
+        elif response.status_code == 401:
+            self.request_token()
+            header = self.header_builder()
+            response = requests.post(url, headers=header)
+        elif response.status_code == 500:
+            raise InternalError()
+        return response.json()['clientID'], response.json()['clientSecret']
+
 
     # TODO: update client's owner
     def update_owner(self, clientID, owner):
-        pass
+        """
+        Parameters from path
+        ----------
+        clientID : str
+            Client Id of the application.
+
+        Parameters from query
+        ----------
+        owner : str
+            (Optional) Username of owner.
+
+        Returns
+        -------
+        str
+            Success message.
+
+        """
+        url = Client.base_URL + 'apps/' + clientID
+        data = {'owner': owner}
+        response = requests.put(url, data)
+        if response.status_code == 401:
+            self.request_token()
+            response = requests.put(url, data)
+        elif response.status_code == 500:
+            raise InternalError()
+        elif response.status_code == 404:
+            raise ResourceNotFoundError(response.json()['message'])
+        return response.json()['message']
+
 
     # TODO: update client's permissionLevel
     def update_permission(self, clientID, permissionLevel):
-        pass
+        """
+        Parameters from path
+        ----------
+        clientID : str
+            Client Id of the application.
+
+        Parameters from query
+        ----------
+        permissionLevel : str
+            (Optional) Permission level of client.
+
+        Returns
+        -------
+        str
+            Success message.
+
+        """
+        url = Client.base_URL + 'apps/' + clientID
+        data = {'permissionLevel': permissionLevel}
+        response = requests.put(url, data)
+        if response.status_code == 401:
+            self.request_token()
+            response = requests.put(url, data)
+        elif response.status_code == 500:
+            raise InternalError()
+        elif response.status_code == 404:
+            raise ResourceNotFoundError(response.json()['message'])
+        return response.json()['message']
+
 
     # TODO: get clientID by owner
     def client_ids_by_owner(self, owner):
-        pass
+        """
+        Parameters
+        ----------
+        owner : str
+            Username of the owner of the client application.
+
+        Returns
+        -------
+        array of str
```

Comment:
for python it should be of type `list`. I don't know if and how you could document something representing an list of strings, since python list can be heterogenous.